pkg/ignition/enrich.go

```diff
@@ -0,0 +1,157 @@
+					},
+					FileEmbedded1: ignitionTypes.FileEmbedded1{
+						Mode: ptr.To(0644),
+						Contents: ignitionTypes.FileContents{
+							Source: fmt.Sprintf("data:,%s", e.Hostname),
+						},
+					},
+				},
+				{
+					Node: ignitionTypes.Node{
+						Filesystem: "root",
+						Path:       "/etc/proxmox-env",
+						Overwrite:  ptr.To(true),
+					},
+					FileEmbedded1: ignitionTypes.FileEmbedded1{
+						Mode: ptr.To(420),
+						Contents: ignitionTypes.FileContents{
+							Source: fmt.Sprintf("data:,%s", e.getProxmoxEnvContent()),
+						},
+					},
+				},
+			},
+		},
+		Networkd: ignitionTypes.Networkd{
+			Units: []ignitionTypes.Networkdunit{},
+		},
+		Systemd: ignitionTypes.Systemd{
+			Units: []ignitionTypes.Unit{
+				{
+					Name:   "systemd-resolved.service",
+					Enable: true,
+				},
+			},
+		},
+	}
+
+	for i, net := range e.Network {
+		ign.Networkd.Units = append(ign.Networkd.Units, ignitionTypes.Networkdunit{
+			Name:     fmt.Sprintf("%d%d-eth%d.network", i, i, i),
+			Contents: getNetworkdUnit(net),
+		})
+	}
+
+	return ign
+}
+
+func getNetworkdUnit(net cloudinit.NetworkConfigData) string {
+	str := fmt.Sprintf("[Match]\nMACAddress=%s\n\n[Link]\nName=%s\n\n[Network]\nAddress=%s\nGateway=%s\n", net.MacAddress, net.Name, net.IPAddress, net.Gateway)
+	if net.IPV6Address != "" {
+		str += fmt.Sprintf("Address=%s\n", net.IPV6Address)
+		str += fmt.Sprintf("Gateway=%s\n", net.Gateway6)
+	}
+	if net.DNSServers != nil {
+		for _, dns := range net.DNSServers {
+			str += fmt.Sprintf("DNS=%s\n", dns)
+		}
+	}
+	return str
+}
+
+func (e *Enricher) getProxmoxEnvContent() string {
+	content := fmt.Sprintf("COREOS_CUSTOM_HOSTNAME=%s\nCOREOS_CUSTOM_INSTANCE_ID=%s\nCOREOS_CUSTOM_PROVIDER_ID=%s", e.Hostname, e.InstanceID, e.ProviderID)
+	if len(e.Network) > 0 && e.Network[0].IPAddress != "" {
+		content += fmt.Sprintf("\nCOREOS_CUSTOM_PRIVATE_IPV4=%s", e.Network[0].IPAddress)
+	}
+	if len(e.Network) > 0 && e.Network[0].IPV6Address != "" {
+		content += fmt.Sprintf("\nCOREOS_CUSTOM_PRIVATE_IPV6=%s", e.Network[0].IPV6Address)
+	}
+	return url.PathEscape(content)
+}
+
+func buildIgnitionConfig(bootstrapData []byte, enrichConfig *ignitionTypes.Config) ([]byte, string, error) {
+	// We control bootstrapData config, so treat it as strict.
+	ign, _, err := convertToIgnition(bootstrapData, false)
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "converting bootstrap-data to Ignition")
+	}
+
+	var clcWarnings string
+	if enrichConfig != nil {
+		ign = ignition.Append(ign, *enrichConfig)
+	}
+
+	userData, err := json.Marshal(&ign)
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "marshaling generated Ignition config into JSON")
+	}
+
+	return userData, clcWarnings, nil
+}
+
+func convertToIgnition(data []byte, strict bool) (ignitionTypes.Config, string, error) {
+	cfg, reports, err := ignition.Parse(data)
+	if err != nil {
+		return ignitionTypes.Config{}, "", errors.Wrapf(err, "parsing ignition Config")
```

Comment:
```suggestion
		return ignitionTypes.Config{}, "", errors.Wrapf(err, "parsing Ignition config")
```

Reply:
to stay consistent with the previous error messages. I personally would write all errors completely in lower case